geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLRules.scala

```diff
@@ -276,6 +276,11 @@ object SQLRules extends LazyLogging {
 
   def registerOptimizations(sqlContext: SQLContext): Unit = {
 
+    Seq(STContainsRule).foreach { r =>
```

Comment:
what about all the other predicates here: https://github.com/ccri/geomesa/blob/ff5a0d5b223c9f25bd3829c7e8091067b06cef7c/geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLRules.scala#L59-L71

Reply:
i guess those are defined as rules - but maybe they should be? doesn't have to be part of this pr

Reply:
Right. This was a point I was gonna bring up with you. The rule is poorly named. It handles all predicates, (and also some spatial join optimizations) but it kept it's original name through all of that